extensions/wikia/Blogs/BlogArticle.php

```diff
@@ -78,24 +83,21 @@ public function view() {
 	 * @access private
 	 */
 	private function showBlogListing() {
-		global $wgOut, $wgRequest, $wgMemc, $wgParser;
+		$wg = F::app()->wg;
+		$wg->Out->setSyndicated( true );
 
-		/**
-		 * use cache or skip cache when action=purge
-		 */
-		$user    = $this->mTitle->getBaseText();
-		$userMem = $this->mTitle->getPrefixedDBkey();
+		$owner = $this->getBlogOwner();
 		$listing = false;
-		$purge   = $wgRequest->getVal( "action" ) == 'purge';
-		$page    = $wgRequest->getVal( "page", 0 );
-		$offset  = $page * $this->mCount;
+		$page = $wg->Request->getVal( "page", 0 );
 		$blogPostCount = null;
 
-		$wgOut->setSyndicated( true );
+		$memc = $wg->Memc;
+		$memKey = $this->blogListingMemcacheKey( $owner, $page );
 
-		if ( !$purge ) {
-			$cachedValueKey  = $this->blogListingMemcacheKey( $userMem, $page );
-			$cachedValue = $wgMemc->get( $cachedValueKey );
+		 // Use cache unless action=purge was used
+		$useCache = $wg->Request->getVal( 'action' ) != 'purge';
+		if ( $useCache ) {
```

Comment:
You may as well put the condition inside the ```if``` block at this point. I think your comment explains it well enough and there's no point for the temporary variable ```$useCache```.